NeuroID/src/main/java/com/neuroid/tracker/service/NIDJobServiceManager.kt

```diff
@@ -59,13 +76,39 @@ object NIDJobServiceManager {
         NIDSensorHelper.restartSensors()
         jobCaptureEvents?.cancel()
         jobCaptureEvents = createJobServer()
+
+        gyroCadenceJob?.cancel()
+        gyroCadenceJob = createGyroJobServer()
     }
 
     private fun createJobServer(): Job {
         return CoroutineScope(Dispatchers.IO).launch {
             while (userActive && isActive) {
                 delay(5000L)
-                sendEventsNow(NIDLogWrapper())
+                sendEventsNow()
+            }
+        }
+    }
+
+    private fun createGyroJobServer(): Job {
+        return CoroutineScope(Dispatchers.IO).launch {
+            while (NeuroID.isSDKStarted && NeuroID.captureGyroCadence) {
+                delay(NeuroID.GYRO_SAMPLE_INTERVAL)
+
+                val gyroData = NIDSensorHelper.getGyroscopeInfo()
```

Comment:
have to check if sensors are running and restart it not (or have restart check for you). the 30 sec sensor timer might have killed them.